packages/global-styles/src/hooks/use-sync-global-styles-user-config.ts

```diff
@@ -1,21 +1,20 @@
-// import { GlobalStylesContext } from '@wordpress/edit-site/build-module/components/global-styles/context';
-// import { mergeBaseAndUserConfigs } from '@wordpress/edit-site/build-module/components/global-styles/global-styles-provider';
 import { useContext, useEffect } from 'react';
+import { GlobalStylesContext, mergeBaseAndUserConfigs } from '../gutenberg-bridge';
 import type { GlobalStylesObject } from '../types';
 
-const useSyncGlobalStylesUserConfig = ( globalStyles: GlobalStylesObject[], enabled: boolean ) => {
-	// const { user, setUserConfig } = useContext( GlobalStylesContext );
-	// useEffect( () => {
-	// 	if ( ! enabled ) {
-	// 		return;
-	// 	}
-	// 	setUserConfig( () =>
-	// 		globalStyles
-	// 			.filter( Boolean )
-	// 			.reduce( ( prev, current ) => mergeBaseAndUserConfigs( prev, current ), {} )
-	// 	);
-	// }, [ globalStyles, enabled ] );
-	// return user;
+const useSyncGlobalStylesUserConfig = ( globalStyles: GlobalStylesObject[] ) => {
+	const { user, setUserConfig } = useContext( GlobalStylesContext );
+	useEffect( () => {
+		setUserConfig( () =>
+			globalStyles
+				.filter( Boolean )
+				.reduce(
+					( prev, current ) => mergeBaseAndUserConfigs( prev, current ),
+					{} as GlobalStylesObject
+				)
+		);
+	}, [ globalStyles ] );
```

Comment:
Is there a chance `globalStyles` will be a new object on every re-render and `setUserConfig` will be called every time? Should we memoize the config merging and call `setUserConfig` only if the memoized version changes?

Reply:
I was mostly thinking out loud and leaving this for your consideration, but if it's handled already, no need to worry about it 👍